### PR TITLE
Seperate components into internal-only category

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

Application Developers trying to build applications with the current Polaris documentation are building poor experiences as they are using components that should not be used for application development. `Navigation` `Frame` `Toast`... will render incorrectly in apps and Application Developers need to use App Bridge API's to correctly build their user interface.

This PR hopes to resolve this by:
1. Move the internal team documentation for components to a new category called `Internal only`
2. Deprecate the components for application developers and direct them to app bridge

The goal of this PR is too:
- Ensure all components can be used by application developers without issue
- Friction is added to components like `<Modal>` `<Toast>` and `<Loading>` to ensure that Application Developers are not incorrectly using these components and going to use app bridge components
- Ensure Shopify employees can still access these components despite them being in a new _more confusing_ location
- Educate users without creating private documentation or shipping breaking changes
- Alerts developers that these components can be removed in a future major version if we decide to move them to a private space

### WHAT is this pull request doing?

- [x] Moves `<ContextualSaveBar>` `<Navigation>` `<TopBar>` `<Toast>` `<Modal>` `<Loading>` `<Frame>` into the Internal (auth required) component category
- [x] New deprecated component pages for `<Navigation>` `<TopBar>` `<Toast>` `<Modal>` `<Loading>` `<Frame>` and directs users to App Bridge API's when they exist
- [x] Adds banners and documentation for deprecated components
- [x] Hides internal only components from search and components landing page
- [x] Updates paths of component images
- [x] Add authentication for `components/internal-only/**/*`
- [x] Redirect component links in shopify.dev and polaris.shopify.com to `internal-only` path

### CODE REVIEW

File: polaris.shopify.com/content/components/deprecated/frame.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Frame
 shortDescription: Creates the structure of the Shopify admin. All of the main sections of the admin are nested in the frame.
-category: Utilities
+category: Deprecated
```
Comment (jesstelford) [inline polaris.shopify.com/content/components/deprecated/frame.mdx:4]: Why are these marked as `Deprecated`? Shouldn't it be `Internal`?

Technically the components aren't deprecated, they're still being consumed and have stable APIs, we just want to limit who consumes them to a single external dependant.

File: polaris.shopify.com/content/components/deprecated/top-bar.mdx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - navbar
+  - brand
+  - search
+  - user
+  - menu
+  - logo
+examples:
+  - fileName: top-bar-default.tsx
+    title: Default
+    description: Use to provide structure for the top of an application. Style the top bar component using the app provider component with a theme. Providing just the `background` key for the top bar component theme will result in intelligent defaults being set for complementary colors with contrasting text.
```
Comment (alex-page) [inline polaris.shopify.com/content/components/deprecated/top-bar.mdx:22]:
```suggestion
    description: Provide structure for the top of an application. Style the top bar component using the app provider component with a theme. Providing just the `background` key for the top bar component theme will result in intelligent defaults being set for complementary colors with contrasting text.
```

File: polaris.shopify.com/content/components/deprecated/frame.mdx (hunk 1 above)
Comment (alex-page) [reply polaris.shopify.com/content/components/deprecated/frame.mdx:4]: I have two files now for each component. One is deprecated which is visible to 3P developers. The other is in internal only. This felt like the best solution to give internal developers the documentation they need while forcefully redirecting application developers to use app bridge.

**Review by craigbrunner — APPROVED** [review]

File: polaris.shopify.com/content/components/internal-only/frame.mdx [hunk 3]
```diff
@@ -43,17 +49,17 @@ The frame component, while not visible in the user interface itself, provides th
 For the best experience when creating an application frame, use the following components:
 
 - [Top bar](https://polaris.shopify.com/components/top-bar)
-- [Navigation](https://polaris.shopify.com/components/navigation/navigation)
+- [Navigation](https://polaris.shopify.com/components/deprecated/navigation)
 - [Contextual save bar](https://polaris.shopify.com/components/contextual-save-bar)
-- [Toast](https://polaris.shopify.com/components/feedback-indicators/toast)
+- [Toast](https://polaris.shopify.com/components/deprecated/toast)
 - [Loading](https://polaris.shopify.com/components/loading)
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/frame.mdx:55]: These should all point to the `/internal-only/` equivalents I believe.

File: polaris.shopify.com/content/components/internal-only/contextual-save-bar.mdx [hunk 4]
```diff
@@ -48,7 +50,7 @@ The contextual save bar component should:
 - Become visible when a form on the page has unsaved changes
 - Be used to save or discard in-progress changes
 - Provide brief and helpful context on the nature of in-progress changes
-- Save all changes on the page. Avoid scenarios where multiple forms on a single page can be edited at the same time. If specific sections of a page need to be independently editable, use an Edit button to launch a [modal dialog](https://polaris.shopify.com/components/overlays/modal) for each section where changes can be made and saved.
+- Save all changes on the page. Avoid scenarios where multiple forms on a single page can be edited at the same time. If specific sections of a page need to be independently editable, use an Edit button to launch a [modal dialog](https://polaris.shopify.com/components/deprecated/modal) for each section where changes can be made and saved.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/contextual-save-bar.mdx:53]: This modal path should be `/internal-only/modal`.

File: polaris.shopify.com/content/components/internal-only/frame.mdx [hunk 5]
```diff
@@ -23,17 +23,23 @@ examples:
...
+hideFromNav: true
+internalOnly: true
 ---
 
 # {frontmatter.title}
 
 <Lede>
 
-The frame component, while not visible in the user interface itself, provides the structure for an application. It wraps the main elements and houses the primary [navigation](https://polaris.shopify.com/components/navigation/navigation), [top bar](https://polaris.shopify.com/components/top-bar), [toast](https://polaris.shopify.com/components/feedback-indicators/toast), and [contextual save bar](https://polaris.shopify.com/components/contextual-save-bar) components.
+The frame component, while not visible in the user interface itself, provides the structure for an application. It wraps the main elements and houses the primary [navigation](https://polaris.shopify.com/components/deprecated/navigation), [top bar](https://polaris.shopify.com/components/top-bar), [toast](https://polaris.shopify.com/components/deprecated/toast), and [contextual save bar](https://polaris.shopify.com/components/contextual-save-bar) components.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/frame.mdx]: These should all point to the `/interanl-only/` paths I believe.

File: polaris.shopify.com/content/components/internal-only/frame.mdx (hunk 3 above)
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/frame.mdx]: These should all point to `/internal-only/`.

File: polaris.shopify.com/content/components/internal-only/frame.mdx [hunk 6]
```diff
@@ -43,17 +49,17 @@ The frame component, while not visible in the user interface itself, provides th
...
+- [Toast](https://polaris.shopify.com/components/deprecated/toast)
 - [Loading](https://polaris.shopify.com/components/loading)
 
 ---
 
 ## Related components
 
 - To display the navigation component on small screens, to provide search and a user menu, or to style the [frame](https://polaris.shopify.com/components/frame) component to reflect an application’s brand, use the [top bar](https://polaris.shopify.com/components/top-bar) component.
-- To display the primary navigation within the frame of an application, use the [navigation](https://polaris.shopify.com/components/navigation/navigation) component.
+- To display the primary navigation within the frame of an application, use the [navigation](https://polaris.shopify.com/components/deprecated/navigation) component.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/frame.mdx]: `/internal-only/`

File: polaris.shopify.com/content/components/internal-only/frame.mdx [hunk 7]
```diff
@@ -43,17 +49,17 @@ The frame component, while not visible in the user interface itself, provides th
...
 - To tell merchants their options once they have made changes to a form on the page use the [contextual save bar](https://polaris.shopify.com/components/contextual-save-bar) component.
-- To provide quick, at-a-glance feedback on the outcome of an action, use the [toast](https://polaris.shopify.com/components/feedback-indicators/toast) component.
+- To provide quick, at-a-glance feedback on the outcome of an action, use the [toast](https://polaris.shopify.com/components/deprecated/toast) component.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/frame.mdx]: `/internal-only/`

File: polaris.shopify.com/content/components/internal-only/loading.mdx [hunk 8]
```diff
@@ -22,6 +24,12 @@ The loading component is used to indicate to merchants that a page is loading or
 
 </Lede>
 
+<StatusBanner status={frontmatter.status}>
+  This component is no longer supported. Please use the [App Bridge Loading
+  API](https://shopify.dev/docs/api/app-bridge-library/reference/loading)
+  instead.
+</StatusBanner>
+
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/loading.mdx]: Can this safely be removed? AFAIK it doesn't output due to `frontmatter.status` being falsey here.

File: polaris.shopify.com/content/components/internal-only/modal.mdx [hunk 9]
```diff
@@ -67,6 +69,11 @@ Modals are overlays that require merchants to take an action before they can con
 
 </Lede>
 
+<StatusBanner status={frontmatter.status}>
+  This component is no longer supported. Please use the [App Bridge Modal
+  API](https://shopify.dev/docs/api/app-bridge-library/reference/modal) instead.
+</StatusBanner>
+
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/modal.mdx]: Can this safely be removed? AFAIK it doesn't output due to `frontmatter.status` being falsey here.

File: polaris.shopify.com/content/components/internal-only/navigation.mdx [hunk 10]
```diff
@@ -58,6 +60,13 @@ The navigation component is used to display the primary navigation in the sideba
 
 </Lede>
 
+<StatusBanner status={frontmatter.status}>
+  This component is no longer supported. Please use the [App Bridge Navigation
+  Menu
+  API](https://shopify.dev/docs/api/app-bridge-library/reference/navigation-menu)
+  instead.
+</StatusBanner>
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/navigation.mdx]: Can this safely be removed? AFAIK it doesn't output due to `frontmatter.status` being falsey here.

File: polaris.shopify.com/content/components/internal-only/toast.mdx [hunk 11]
```diff
@@ -41,6 +43,11 @@ The toast component is a non-disruptive message that appears at the bottom of th
 
 </Lede>
 
+<StatusBanner status={frontmatter.status}>
+  This component is no longer supported. Please use the [App Bridge Toast
+  API](https://shopify.dev/docs/api/app-bridge-library/reference/toast) instead.
+</StatusBanner>
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/toast.mdx]: Can this safely be removed? AFAIK it doesn't output due to `frontmatter.status` being falsey here.

File: polaris.shopify.com/content/components/internal-only/top-bar.mdx [hunk 12]
```diff
@@ -31,6 +33,10 @@ The top bar is a header component that allows merchants to search, access menus,
 
 </Lede>
 
+<StatusBanner status={frontmatter.status}>
+  This component is no longer supported.
+</StatusBanner>
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/top-bar.mdx]: Can this safely be removed? AFAIK it doesn't output due to `frontmatter.status` being falsey here.

File: polaris.shopify.com/content/components/internal-only/top-bar.mdx [hunk 13]
```diff
@@ -46,7 +52,7 @@ The top bar component must be passed to the [frame](https://polaris.shopify.com/
 The top bar component should:
 
 - Not provide global navigation for an application
-  - Use the [navigation component](https://polaris.shopify.com/components/navigation/navigation) instead
+  - Use the [navigation component](https://polaris.shopify.com/components/deprecated/navigation) instead
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/top-bar.mdx]: Path should be `/internal-only/`

File: polaris.shopify.com/content/components/internal-only/top-bar.mdx [hunk 14]
```diff
@@ -147,7 +153,7 @@ A text field component that is tailor-made for a search use-case.
 ## Related components
 
 - To provide the structure for the top bar component, as well as the primary navigation use the [frame](https://polaris.shopify.com/components/frame) component.
-- To display the primary navigation within the frame of an application, use the [navigation](https://polaris.shopify.com/components/navigation/navigation) component.
+- To display the primary navigation within the frame of an application, use the [navigation](https://polaris.shopify.com/components/deprecated/navigation) component.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/top-bar.mdx]: `/internal-only/`

File: polaris.shopify.com/content/components/internal-only/top-bar.mdx [hunk 15]
```diff
@@ -147,7 +153,7 @@ A text field component that is tailor-made for a search use-case.
...
 - To tell merchants their options once they have made changes to a form on the page use the [contextual save bar](https://polaris.shopify.com/components/contextual-save-bar) component.
-- To provide quick, at-a-glance feedback on the outcome of an action, use the [toast](https://polaris.shopify.com/components/feedback-indicators/toast) component.
+- To provide quick, at-a-glance feedback on the outcome of an action, use the [toast](https://polaris.shopify.com/components/deprecated/toast) component.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/internal-only/top-bar.mdx]: `/internal-only/`

File: polaris.shopify.com/content/components/navigation/fullscreen-bar.mdx [hunk 16]
```diff
@@ -40,5 +40,5 @@ The Fullscreen bar component should:
 
 ## Related components
 
-- To provide quick, at-a-glance feedback on the outcome of an action, use the [toast](https://polaris.shopify.com/components/feedback-indicators/toast) component.
+- To provide quick, at-a-glance feedback on the outcome of an action, use the [toast](https://polaris.shopify.com/components/deprecated/toast) component.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/components/navigation/fullscreen-bar.mdx]: Can we change this to `use the [App Bridge toast API](https://shopify.dev/docs/api/app-bridge-library/reference/toast).`

File: polaris.shopify.com/content/content/voice-and-tone.mdx [hunk 17]
```diff
@@ -205,7 +205,7 @@ While we don’t need to celebrate the accomplishment, we should recognize that
 
 #### Do
 
-If this is a task people do regularly, make it clear that the step is complete in a simple and non-intrusive way, like a [toast](/components/feedback-indicators/toast). Consider ways to confirm completion without words or messaging.
+If this is a task people do regularly, make it clear that the step is complete in a simple and non-intrusive way, like a [toast](/components/deprecated/toast). Consider ways to confirm completion without words or messaging.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/content/voice-and-tone.mdx]: Can we change this to something like:

> If this is a task people do regularly, make it clear that the step is complete in a simple and non-intrusive way, like via the [App Bridge toast](https://shopify.dev/docs/api/app-bridge-library/reference/toast). Consider ways to confirm completion without words or messaging.

File: polaris.shopify.com/content/design/interaction-states.mdx [hunk 18]
```diff
@@ -80,6 +80,6 @@ Provide merchants with cues as to what the interface will do if they interact wi
 
 **Use feedback indicators** like the [progress bar](https://polaris.shopify.com/components/progress-bar) component or the [spinner](https://polaris.shopify.com/components/spinner) component to let them know that the interface received their request. If appropriate, you can also provide added information about what or how long it will take to complete.
 
-**For non-disruptive feedback** on the outcome of an action, use the [toast](https://polaris.shopify.com/components/feedback-indicators/toast) component.
+**For non-disruptive feedback** on the outcome of an action, use the [toast](https://polaris.shopify.com/components/deprecated/toast) component.
```
Comment (themarkappleby) [inline polaris.shopify.com/content/design/interaction-states.mdx]: ...use the [App Bridge toast API](https://shopify.dev/docs/api/app-bridge-library/reference/toast).

File: polaris.shopify.com/src/components/Frame/Frame.module.scss [hunk 19]
```diff
@@ -73,6 +73,10 @@ $breakpointNav: $breakpointTablet;
     flex-direction: column;
 
     > li {
+      > ul > li + li {
+        padding-top: 1rem;
```
Comment (themarkappleby) [inline polaris.shopify.com/src/components/Frame/Frame.module.scss]: This is causing some unintended spacing in other parts of the navigation.

Screenshot 2024-01-12 at 2 57 17 PM

File: polaris.shopify.com/src/components/StatusBanner/StatusBanner.tsx [hunk 20]
```diff
@@ -11,7 +11,10 @@ function StatusBanner({status, children}: Props) {
   return (
     status && (
       <div className={styles.StatusBanner} data-value={status.toLowerCase()}>
-        <h2>{uppercaseFirst(status)}</h2>
+        <h2>
+          {uppercaseFirst(status)}
+          {status === 'Internal' ? ' only' : null}
```
Comment (themarkappleby) [inline polaris.shopify.com/src/components/StatusBanner/StatusBanner.tsx]: Is this necessary? I can't find any instances of `frontmatter.status` being set to `Internal`. 

12-05-w47u5-w2tqa

**Review by themarkappleby — CHANGES_REQUESTED** [review]
This is amazing, thanks so much @alex-page! Just a few minor questions/comments from me 👍

File: polaris.shopify.com/content/components/internal-only/frame.mdx (hunk 5 above)
Comment (alex-page) [reply polaris.shopify.com/content/components/internal-only/frame.mdx]: Ahhhh yes. Any of the internal only paths that reference other internal only components I will update. Public paths I will keep pointing to deprecated. Thanks @themarkappleby.

**Pull request merged**